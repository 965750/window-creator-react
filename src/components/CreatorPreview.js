import React, { Component } from 'react'
import Window from '../components/Window'
import PreviewSwitcher from './PreviewSwitcher'
import styles from './CreatorPreview.module.scss'
import WindowRulers from './WindowRulers'
import { connect } from "react-redux";

class CreatorPreview extends Component {
    state = {
        interiorPreview: true
    }

    onSwitchPreview = (isSwiched) => {
        this.setState({
            interiorPreview: isSwiched
        })
    }

    render () {
        return (
            <div className={`${styles.preview} w-2/3 border border-mercury relative`}>
                <PreviewSwitcher interiorPreview={this.state.interiorPreview} onSwitchPreview={this.onSwitchPreview} />
                <div className={`${this.state.interiorPreview ? styles.cityBackground : ''} absolute flex`}>
                    <WindowRulers doorType={this.props.window.doorType} height={this.props.window.height} width={this.props.window.width} />
                    { [...Array(this.props.window.doorType)].map(() => {
                        return (
                            <Window height={this.props.window.height} width={this.props.window.width}/>
                        )
                    }) }
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        window: state.creator.window
    }
}

export default connect(mapStateToProps)(CreatorPreview)
