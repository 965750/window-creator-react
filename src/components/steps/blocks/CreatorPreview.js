import { connect } from 'react-redux'

import cityImage from '../../../assets/city.jpg'
import PreviewSwitcher from './PreviewSwitcher'
import PropTypes from 'prop-types'
import React, { Component } from 'react'
import styles from './CreatorPreview.module.scss'
import Window from './Window'
import WindowRulers from './WindowRulers'

class CreatorPreview extends Component {
  state = {
    interiorPreview: true,
  }

  onSwitchPreview = (isSwiched) => {
    this.setState({
      interiorPreview: isSwiched,
    })
  }

  render() {
    return (
      <div
        className={`${styles.preview} ${
          this.state.interiorPreview ? styles.preview__background : ''
        } w-full md:w-2/3 border border-mercury relative`}
      >
        <PreviewSwitcher
          interiorPreview={this.state.interiorPreview}
          onSwitchPreview={this.onSwitchPreview}
        />
        <div className={`${styles.city__window} absolute flex`}>
          {this.state.interiorPreview ? (
            <div className="overflow-hidden absolute w-full h-full">
              <img
                alt="city"
                src={cityImage}
                className={`${styles.city__background} absolute bottom-0`}
              />
            </div>
          ) : null}
          <WindowRulers window={this.props.window} />
          {[...Array(this.props.window.doorType)].map((single, index) => (
            <Window
              innerWidth={window.innerWidth}
              key={index} // eslint-disable-line
              window={this.props.window}
            />
          ))}
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  window: state.creator.window,
})

CreatorPreview.propTypes = {
  window: PropTypes.shape({
    color: PropTypes.string,
    width: PropTypes.number,
    height: PropTypes.number,
    doorType: PropTypes.number,
    rows: PropTypes.array,
    columns: PropTypes.array,
  }),
}

export default connect(mapStateToProps)(CreatorPreview)
