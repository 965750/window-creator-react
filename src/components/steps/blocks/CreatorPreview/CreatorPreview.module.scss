.preview {
    height: 500px;

    @media screen and (max-width: 768px) {
        height: 310px;
    }

    &__background {
        background-image: url('../../../../assets/room.jpg');
        background-position: -150px -24px;
        background-size: 820px 571px;
        background-repeat: no-repeat;

        @media screen and (max-width: 768px) {
            background-position: -140px -4px;
            background-size: 517px 360px;
        }
    }
}

.city {

    &__window {
        bottom: 99px;
        left: 120px;

        @media screen and (max-width: 768px) {
            bottom: 45px;
            left: 64px;
        }
    }

    &__background {
        min-width: 530px;
    }
}