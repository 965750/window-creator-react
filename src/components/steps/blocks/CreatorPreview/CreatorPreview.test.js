import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react'

import CreatorPreview from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk'
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
  creator: {
    window: {
      color: '#000',
      width: 110,
      height: 230,
      doorType: 2,
      rows: [0],
      columns: [0],
    },
  }
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('CreatorPreview', () => {
  let wrapper
  const initialProps = {}

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <CreatorPreview {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
