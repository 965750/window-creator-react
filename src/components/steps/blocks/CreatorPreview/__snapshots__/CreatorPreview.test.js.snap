// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorPreview should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="preview preview__background w-full md:w-2/3 border border-mercury relative"
  >
    <div
      class="flex cursor-pointer text-center absolute top-10 right-20"
    >
      <button
        class="sideBtn border-2 bg-white border-gullGray"
        data-testid="previewSwitcher3d"
      >
        3D
      </button>
      <button
        class="sideBtn border bg-botticelli opacity-75 border-gullGray"
        data-testid="previewSwitcher2d"
      >
        2D
      </button>
    </div>
    <div
      class="city__window absolute flex"
    >
      <div
        class="overflow-hidden absolute w-full h-full"
      >
        <img
          alt="city"
          class="city__background absolute bottom-0"
          src="city.jpg"
        />
      </div>
      <div>
        <div
          class="ruler--vertical absolute -left-30 md:-left-40 h-full border-t-2 border-scorpion border-b-2"
        >
          <div
            class="ruler__counter absolute top-half left-half border bg-white text-center"
          >
            <span>
              230
            </span>
          </div>
          <div
            class="bar--vertical h-full bg-scorpion mx-auto"
          />
        </div>
        <div
          class="ruler--horizontal absolute -top-25 md:-top-40 w-full border-l-2 border-scorpion border-r-2"
        >
          <div
            class="ruler__counter absolute top-half left-half border bg-white text-center"
          >
            <span>
              220
            </span>
          </div>
          <div
            class="bar--horizontal w-full bg-scorpion"
          />
        </div>
        <div
          class="flex"
        >
          <div
            class="ruler--horizontal absolute -bottom-25 md:-bottom-40 w-1/2 border-l-2 border-scorpion border-r-2"
          >
            <div
              class="ruler__counter absolute top-half left-half border bg-white text-center"
            >
              <span>
                110
              </span>
            </div>
            <div
              class="bar--horizontal w-full bg-scorpion"
            />
          </div>
          <div
            class="ruler--horizontal absolute -bottom-25 md:-bottom-40 w-1/2 right-0 border-l-2 border-scorpion border-r-2"
          >
            <div
              class="ruler__counter absolute top-half left-half border bg-white text-center"
            >
              <span>
                110
              </span>
            </div>
            <div
              class="bar--horizontal w-full bg-scorpion"
            />
          </div>
        </div>
      </div>
      <div
        class="h-full z-20"
        style="width: 110px; height: 230px;"
      >
        <table
          class="window w-full h-full"
          style="border-color: #000;"
        >
          <tbody>
            <tr
              class="window__rows"
              style="border-color: #000;"
            >
              <td
                class="window__columns"
                style="border-color: #000;"
              >
                 
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="h-full z-20"
        style="width: 110px; height: 230px;"
      >
        <table
          class="window w-full h-full"
          style="border-color: #000;"
        >
          <tbody>
            <tr
              class="window__rows"
              style="border-color: #000;"
            >
              <td
                class="window__columns"
                style="border-color: #000;"
              >
                 
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
